pkg/kubectl/cmd/cp.go

```diff
@@ -453,8 +454,24 @@ func untarAll(reader io.Reader, destFile, prefix string) error {
 }
 
 func getPrefix(file string) string {
+	segs := strings.Split(file, "/")
+	prefix := file
+
+	lastIdx := -1
+	for i := len(segs) - 1; i > 0; i-- {
+		if len(segs[i]) == 0 {
```

Comment:
can you describe the purpose of this function, and of this change?

Reply:
it modified the remote path to just return the prefix without the filename, i.e. for a string "foo/bar/baz.exe" it returned "foo/bar/". I realize this changes existing behavior, particularly when attempting to copy a remote file under a different filename locally (copying remote file "/foo/bar/baz" to local file "buz" would now just create a directory "buz" and place file "baz" in it). Will revert this